text/078-tbc-title.md

```diff
@@ -0,0 +1,508 @@
+
+This lead us (UI team) to do initial research at the start of 2022 into what frameworks exist (based on suggestions from the Wagtail community). See [Lightweight JavaScript Framework research](https://github.com/lb-/bakerydemo/blob/ui-experiments/lightweight-frontend-framework-investigation-2022.md).
+
+This also lead to various rounds of feedback about what approaches could work with a leaning towards Stimulus. See the following links:
+
+- [main discussion sub-tread on a lightweight framework](https://github.com/wagtail/wagtail/discussions/7689#discussioncomment-2037913)
+- [page editor initial discussion on a lightweight framework](https://github.com/wagtail/wagtail/discussions/7739#discussioncomment-2022909)
+- [page editor discussion resulting in suggestions for frameworks](https://github.com/wagtail/wagtail/discussions/7739#discussioncomment-1926410)
+- [feedback from Matt](https://github.com/wagtail/wagtail/discussions/7689#discussioncomment-2775453)
+
+## Specification
+
+### Initial Implementation Overview
+
+- Install Stimulus `npm install @hotwired/stimulus --save`
+- Due to Stimulus being an ES6 module and Wagtail's codebase transpiling to ES5, TypeScript/Webpack and Jest will need to be configured to ensure that this library within node modules is to be transpiled also.
+- Set up a file `client/src/includes/stimulus.ts` which will contain the Stimulus application set up and register of initial controllers, it should export a named export `initStimulus` which will allow for control over when the Application is set up.
+- The dispatched events and event listeners should be set up as described in the documentation snippets below.
+- This file should have an automatic way to pull in any Controllers in the folder `client/src/controllers/` such that a controller file named `TagFieldController.ts` with a named export `{ TagFieldController }` will automatically be registered under the prefixed namespace `w-tag-field` for its identifier. The inspiration for this approach can be found here https://github.com/hotwired/stimulus-webpack-helpers
+- There should be a base `AbstractController` that uses [TypeScript's Abstract class](https://www.typescriptlang.org/docs/handbook/2/classes.html#abstract-classes-and-members) system to provide any base Wagtail methods, core overrides to Stimulus' classes and ensure that derived classes adhere to any expected methods.
+- There should be a simple way to register custom controllers, aimed at those using Wagtail (not core developers), without any prior knowledge of how JavaScript classes work or needing any kind of build tool. The inspiration for this is [Alpine.js' bind approach](https://alpinejs.dev/globals/alpine-bind) and [StackOverflow's Stimulus JS library controller creation](https://stackoverflow.design/product/guidelines/javascript/#creating-your-own-stimulus-controllers).
```

Comment:
Do we expect that everyone adding custom JS into Wagtail to use Stimulus JS?

Reply:
Not at all. Same as we don't expect everyone to use React or jQuery but the ability will be there nonetheless. 

Maybe I can reword this better.